test/perf/main.go

```diff
@@ -0,0 +1,304 @@
+		return nil, fmt.Errorf("failed initializing printer: %w", err)
+	}
+	return p, nil
+}
+
+type csvPrinter struct {
+	writer   *csv.Writer
+	workload string
+}
+
+func (c *csvPrinter) init() error {
+	file, err := os.Create(
+		fmt.Sprintf("./performance-test-results-%v.csv", time.Now().Format("2006-01-02-15-04-05")),
+	)
+	if err != nil {
+		return fmt.Errorf("failed creating file: %w", err)
+	}
+	c.writer = csv.NewWriter(file)
+	err = c.writer.Write([]string{
+		"workload",
+		"total records",
+		"rec/s (Conduit)",
+		"ms/record (Conduit)",
+		"rec/s (pipeline)",
+		"bytes/s",
+		"measured_at",
+	})
+	if err != nil {
+		return err
+	}
+	c.writer.Flush()
+	return nil
+}
+
+func (c *csvPrinter) print(m metrics) error {
+	err := c.writer.Write([]string{
+		c.workload,
+		fmt.Sprintf("%v", m.count),
+		fmt.Sprintf("%v", m.recordsPerSec),
+		m.msPerRecStr(),
+		fmt.Sprintf("%v", m.pipelineRate),
+		fmt.Sprintf("%v", m.bytesPerSec),
+		fmt.Sprintf("%v", m.measuredAt.Format(time.RFC3339)),
+	})
+	if err != nil {
+		return err
+	}
+	c.writer.Flush()
+	return nil
+}
+
+type consolePrinter struct {
+	renderer *glamour.TermRenderer
+	workload string
+}
+
+func (c *consolePrinter) init() error {
+	r, _ := glamour.NewTermRenderer(
+		// detect background color and pick either the default dark or light theme
+		glamour.WithAutoStyle(),
+		glamour.WithWordWrap(200),
+	)
+	c.renderer = r
+	return nil
+}
+
+func (c *consolePrinter) print(m metrics) error {
+	in := `
+| workload | total records | rec/s (Conduit) | ms/record (Conduit) | rec/s (pipeline) | bytes/s | measured at |
+|----------|---------------|-----------------|---------------------|------------------|---------|-------------|
+| %v       | %v            | %v              | %v                  | %v               | %v      | %v          |
+
+
+`
+	in = fmt.Sprintf(
+		in,
+		c.workload,
+		m.count,
+		m.recordsPerSec,
+		m.msPerRecStr(),
+		m.pipelineRate,
+		m.bytesPerSec,
+		m.measuredAt.Format(time.RFC3339),
+	)
+	out, err := c.renderer.Render(in)
+	if err != nil {
+		return fmt.Errorf("failed rendering output: %w", err)
+	}
+	fmt.Print(out)
+	return nil
+}
+
+type collector struct {
+	first metrics
+}
+
+func newCollector() (collector, error) {
+	c := collector{}
+	err := c.init()
+	if err != nil {
+		return collector{}, fmt.Errorf("failed initializing collector: %w", err)
+	}
+	return c, err
+}
+
+func (c *collector) init() error {
+	first, err := c.collect()
+	if err != nil {
+		return err
+	}
+	c.first = first
+	return nil
+}
+
+func (c *collector) collect() (metrics, error) {
+	metricFamilies, err := c.getMetrics()
+	if err != nil {
+		return metrics{}, fmt.Errorf("failed getting metrics: %v", err)
+	}
+
+	m := metrics{}
+	count, totalTime, err := c.getPipelineMetrics(metricFamilies)
+	if err != nil {
+		fmt.Printf("failed getting pipeline metrics: %v", err)
+		os.Exit(1)
+	}
+	m.count = count
+	m.recordsPerSec = float64(count) / totalTime
+	m.msPerRec = (totalTime / float64(count)) / 1000
```

Comment:
```suggestion
	m.msPerRec = (totalTime / float64(count)) * 1000
```

Reply:
I was testing the tests just a bit before, and was just wondering: why so many zeros? : )